### ISSUE TITLE
Creatures can attack through walls when attacking Sonar targets

### ISSUE THREAD

**Issue by Kaillera** [Regalis11/Barotrauma#2119]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Navigation Terminals and players with Handheld Sonar/Sonar Beacon can be attacked through walls by creatures from outside the sub.

**Steps To Reproduce**
Probably the best spot is the Kastrull's command room. Boot up a sonar beacon and stand near the windows or climb close to the ceiling. Mudraptors and Crawlers can bite you through the walls.

**Version**
~0.9.702

**Additional information**
Ow...

**Comment by itchyOwl** [comment]
Should now be fixed.

The issue was not related to sonar beacons or handheld sonars. Actually the ai target of the handheld sonar doesn't seem to work at all right now (-> I'll fix it). My theory is that having a sonar beacon in hand made the creature target you, but it also had a wall target because it was outside. However that wall probably did not have an ai target (when I wrote the code all walls were supposed to have an ai target, but an exception to this was made later). So the distance to the target was calculated properly to the wall target but the attack was applied to the selected target (i.e. you). This is now fixed in Regalis11/Barotrauma-development@c411abe.

It is also possible that in some rare cases the attack limbs of the creatures were inside the wall colliders, which would make the raycast fail. This was fixed in Regalis11/Barotrauma-development@ddf5a61.
